src/components/List.js

```diff
@@ -1,12 +1,54 @@
 import React from 'react';
 import useFirestore from '../hooks/useFirestore';
+import { db } from '../lib/firebase';
 
 const List = ({ token }) => {
   const { docs } = useFirestore(token);
+
+  const checkPurchasedDate = (purchasedDate) => {
+    if (purchasedDate === null) {
+      return false;
+    }
+    const today = new Date();
+    let daySincePurchased = purchasedDate.toDate();
+
+    daySincePurchased.setDate(daySincePurchased.getDate() + 1); // Add one day to the last purchased date
+
+    // If today > daySincePurchased at least 24 hours have passed, return false to uncheck box
+    if (
+      Date.parse(today.toString()) >= Date.parse(daySincePurchased.toString())
+    ) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleCheckbox = async (event) => {
+    const currentDate = new Date();
+    const queryCollection = await db.collection(token).doc(event.target.id);
```

Comment:
Nice use of the id field on the event to track which item to update 👍

Reply:
I'll second that one! Nice one!